File: 11. Sorting/Selection Sort/SelectionSort.java
public class SelectionSort {

    static void printArray(int arr[]) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    static void selectionSort(int arr[]) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[minIdx];
            arr[minIdx] = temp;
        }
        printArray(arr); 
    }

    public static void main(String[] args) {
        int[] arr = {5, 4, 1, 4, 2};
        selectionSort(arr);
    }
}

File: 11. Sorting/Selection Sort/selectionSort.py
def print_arr(arr):
    print(" ".join(map(str, arr)))

def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    print_arr(arr)

if __name__ == "__main__":
    arr = [5, 4, 1, 4, 2]
    selection_sort(arr)
